detox/src/devices/drivers/ios/tools/AppleSimUtils.js

```diff
@@ -345,12 +345,16 @@ class AppleSimUtils {
     return result;
   }
 
-  async _printLoggingHint(udid, bundleId) {
+  async _assembleLoggingHint(udid, bundleId) {
     const appContainer = await this.getAppContainer(udid, bundleId);
     const CFBundleExecutable = await exec.execAsync(`/usr/libexec/PlistBuddy -c "Print CFBundleExecutable" "${path.join(appContainer, 'Info.plist')}"`);
     const predicate = `process == "${CFBundleExecutable}"`;
-    const command = `/usr/bin/xcrun simctl spawn ${udid} log stream --level debug --style compact --predicate '${predicate}'`;
+    this.loggingHintCommand = `/usr/bin/xcrun simctl spawn ${udid} log stream --level debug --style compact --predicate '${predicate}'`;
+    return this.loggingHintCommand;
+  }
 
+  async _printLoggingHint(udid, bundleId) {
+    const command = this.loggingHintCommand || await this._assembleLoggingHint(udid, bundleId);
```

Comment:
Overall, while I understand and support your good intention, I don't think this code is really valid.

The biggest reason is that the applesimutils class, like DeviceDriver class and unlike Device class is currently state less, and this change breaks maybe stupid but till now consistent logic. 

The smaller but a more practical reason is that this code will be invalid after first next app reinstall (weirdly enough, but app container path is changing all the time, the GUID part in the middle of it).

An even smaller reason which still is valid is that any attempt to launch another app will produce an incorrect debug logging hint. 

I don't recommend to leave it this way, it breaks many things in a subtle way.